Minio/Helper/s3utils.cs

```diff
@@ -80,7 +83,52 @@ internal static string GetPath(string p1, string p2)
 
         }
 
-      
-     
+        // IsValidDomain validates if input string is a valid domain name.
+        public static bool IsValidDomain(string host)
```

Comment:
`isValidEndpoint()` and this function does the same thing.  The logic in `isValidEndpoint()` is correct one as per wikipedia.